### ISSUE TITLE
feathers upgrade fails in events.js ERR_INVALID_ARG_TYPE on mac os

### ISSUE THREAD

**Issue by scottmac991** [feathersjs/feathers#1650]
### Steps to reproduce
$cd project_src_dir

$ls
README.md		package-lock.json	src/
babel.config.js		package.json
node_modules/		public/

$feathers upgrade
events.js:186
      throw er; // Unhandled 'error' event
      ^
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.join (path.js:1040:7)
    at UpgradeGenerator.destinationPath (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:701:30)
    at UpgradeGenerator.writing (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/generator-feathers/generators/upgrade/index.js:30:27)
    at Object.<anonymous> (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:395:23)
    at /usr/local/lib/node_modules/@feathersjs/cli/node_modules/run-async/index.js:25:25
    at new Promise (<anonymous>)
    at /usr/local/lib/node_modules/@feathersjs/cli/node_modules/run-async/index.js:24:19
    at /usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:396:9
    at processImmediate (internal/timers.js:439:21)
Emitted 'error' event on UpgradeGenerator instance at:
    at Immediate.<anonymous> (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:404:18)
    at processImmediate (internal/timers.js:439:21)

### Expected behavior
Should succeed

### Actual behavior
Stack trace posted above

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
$ npm list | grep feathers
├─┬ @feathersjs/authentication-client@4.3.0
│ ├── @feathersjs/commons@4.3.0 deduped
│ ├─┬ @feathersjs/errors@4.3.0
├─┬ @feathersjs/cli@4.2.2
│ ├─┬ @feathersjs/tools@0.1.7
│ │ ├─┬ @feathersjs/jscodeshift@0.5.0
│ ├─┬ generator-feathers@4.1.2
│ │ ├── @feathersjs/jscodeshift@0.5.0 deduped
│ │ ├── @feathersjs/tools@0.1.7 deduped
│ ├─┬ generator-feathers-plugin@1.0.1
├── @feathersjs/commons@4.3.0
├─┬ @feathersjs/feathers@4.3.0
│ ├── @feathersjs/commons@4.3.0 deduped
├─┬ @feathersjs/socketio-client@4.3.0
│ ├─┬ @feathersjs/transport-commons@4.3.0
│ │ ├── @feathersjs/commons@4.3.0 deduped
│ │ ├─┬ @feathersjs/errors@4.3.0
├─┬ feathers-vuex@1.7.0
│ ├─┬ @feathersjs/adapter-commons@2.0.0
│ │ ├── @feathersjs/commons@4.3.0 deduped
│ │ └── @feathersjs/errors@3.3.6 deduped
│ ├── @feathersjs/commons@4.3.0 deduped
│ ├─┬ @feathersjs/errors@3.3.6

**NodeJS version**:
$ node --version
v12.9.0

**Operating System**:
Mac OS Catalina 10.15

**Browser Version**:

**React Native Version**:
NA - using iTerm

**Module Loader**:

**Comment by daffl** [comment]
If you customized your project (which it looks like due to the Babel reference) the upgrade generator may not work as intended. Hard to say if it could still work without the entire project to reproduce though.

**Comment by lexparsimonet** [comment]
I'm experiencing a nearly identical error -  I am faced with this output:
(Updating from Feathers 2.2.3 to 4.3, following https://docs.feathersjs.com/guides/migrating.html) 

$feathers update

events.js:298
      throw er; // Unhandled 'error' event
      ^

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at validateString (internal/validators.js:118:11)
    at Object.join (path.js:1039:7)
    at UpgradeGenerator.destinationPath (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:1018:30)
    at UpgradeGenerator.writing (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/generator-feathers/generators/upgrade/index.js:30:27)
    at Object.<anonymous> (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:655:25)
    at /usr/local/lib/node_modules/@feathersjs/cli/node_modules/run-async/index.js:47:25
    at new Promise (<anonymous>)
    at /usr/local/lib/node_modules/@feathersjs/cli/node_modules/run-async/index.js:24:19
    at self.env.runLoop.add.once.once (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:656:11)
    at processImmediate (internal/timers.js:456:21)
Emitted 'error' event on UpgradeGenerator instance at:
    at Immediate.<anonymous> (/usr/local/lib/node_modules/@feathersjs/cli/node_modules/yeoman-generator/lib/index.js:676:20)
    at processImmediate (internal/timers.js:456:21) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node Version
13.10.0
Operating System
Ubuntu 18.04 (same error on 16.04)

Also using babel, but to transpile the react UI

**Comment by scottmac991** [comment]
Hi @lexparsimonet, after this issue got closed I went back to the drawing board and I was able to resolve it. In my case it was due to module incompatibility amongst all the rapidly evolving featherjs packages after my initial project setup. I needed to ensure that ALL of my feathersjs modules were the latest. These included:

@feathersjs/feathers
@feathersjs/socketio-client
socket.io-client
@feathersjs/authentication-client
...and whatever else you are including

I would advise the following:
1. In your project, identify all of the feathers modules and note their versions
`npm  list (for global) OR npm list -g --depth=0 (for local)`

2. Find the latest available package versions available
`npm show [pkg]`
You might notice there are more up to date versions available

3. install the latest package versions - `npm  install  [pkg]`should download the latest

Let me know if it helps.
